python/postprocessing/modules/common/muonScaleResProducer.py

```diff
@@ -37,54 +36,54 @@ def endJob(self):
 
     def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.out = wrappedOutputTree
-        self.out.branch("Muon_pt_corrected", "F", lenVar="nMuon")
-        if self.is_2017:
-            self.out.branch("Muon_pt_sys_uncert", "F", lenVar="nMuon")
+        self.out.branch("Muon_corrected_pt", "F", lenVar="nMuon")
+        self.out.branch("Muon_correctedUp_pt", "F", lenVar="nMuon")
+        self.out.branch("Muon_correctedDown_pt", "F", lenVar="nMuon")
         self.is_mc = bool(inputTree.GetBranch("GenJet_pt"))
 
     def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         pass
 
     def analyze(self, event):
         muons = Collection(event, "Muon")
+        if self.is_mc:
+            genparticles = Collection(event, "GenPart")
         roccor = self._roccor
-
         if self.is_mc:
             u1 = random.uniform(0.0, 1.0)
             u2 = random.uniform(0.0, 1.0)
-            pt_corr = list(
-                mu.pt * mk_safe(
-                    roccor.kScaleAndSmearMC,
-                    mu.charge, mu.pt, mu.eta, mu.phi, mu.nTrackerLayers, u1, u2
-                ) for mu in muons)
-            if self.is_2017:
-                pt_err = list(
-                    mu.pt * mk_safe(
-                        roccor.kScaleAndSmearMCerror,
-                        mu.charge, mu.pt, mu.eta, mu.phi, mu.nTrackerLayers, u1, u2
-                    ) for mu in muons)
+            pt_corr=[]
```

Comment:
why is "u1" reused for all muons? is that in the prescription? isn't this smearing all muons in an event in a correlated way?

Reply:
Yes you are correct. The u1 random number has to different for all the muons. I will close this PR. Some updates are required.